src/main/scala/actions/Action.scala

```diff
@@ -26,11 +28,13 @@ abstract class ReferenceListAction(keyword: String) extends Action {
     s"$keyword " + (references map (_.toQuery(referenceableMap)) mkString ", ")
 }
 
-case class ReturnReference(reference: ReferenceType, rest: ReferenceType*) extends ReferenceListAction("RETURN")
-case class ReturnDistinct(reference: ReferenceType, rest: ReferenceType*) extends ReferenceListAction("RETURN DISTINCT")
-case class Delete(reference: ReferenceType, rest: ReferenceType*) extends ReferenceListAction("DELETE")
+case class Delete(reference: Reference, rest: Reference*) extends ReferenceListAction("DELETE")
+
+case class ReturnReference(reference: Reference, rest: Reference*) extends ReferenceListAction("RETURN") with ReturnAction
+
+case class ReturnDistinct(reference: Reference, rest: Reference*) extends ReferenceListAction("RETURN DISTINCT") with ReturnAction
```

Comment:
not totally sure, but shouldn't `DISTINCT` be related to the column (`Reference`), not the `RETURN`. For example:
`RETURN n, DISTINCT m`